### DISCUSSION TITLE
DOC Transition to IPV6

### DISCUSSION THREAD

**Discussion by NathanHolcomb** [opendcs/opendcs#168] [General]
Good Morning, 
I wanted to reach out to the OpenDCS community to see if the upcoming IPV4 to IPV6 transition would have any unknown consequences for OpenDCS users on DOC networks and machines. The guidance from OMB for the transition is listed here for anyone interested: [DOC IPV6 policy](https://www.commerce.gov/about/policies/ipv6-policy#:~:text=The%20DOC%20will%20phase%20out,no%20later%20than%20FY%202023%20.) 

OpenDCS appears to accept IPV6 addresses with no length limitation issues so I'm curious if there are any other issues that might need to be mitigated prior to this transition. I also imagine this will be extended beyond DOC in the near future if it already hasn't been and will have impacts on the larger community.

Thanks,
Nathan

**Comment by MikeNeilson** [comment]
I wouldn't expect too many problems in OpenDCS itself. There's likely a few places that do an address validation check that may be IPv4 specific but once identified those would be fairly easy to fix. But we should start making sure any tests of network connectivity also work with IPv6... once we figure out exactly how.

That said, I'd be more concerned about downstream users. For example I still need to request an account for my home lab, but I'm on Comcast at home as an ISP. I don't get an IPv6 address; so my home lab may not be able to connect if only IPv6 connectivity is provided. 

I don't think we've gotten IPv6 address at the office yet either. That could be true of many agencies especially small partners.

So other issues, this piece of text implies I control my ISP:

- All external partners will identify systems that interface with networked Federal information systems and develop plans to migrate all such network interfaces to the use of IPv6.

I don't. That would be the FTC/FCC, and even then barely. So my plan is "wait for my ISP to provide said service."
And at the office there's already a push for IPv6... been going on for years.

Granted I could get an IPv6 tunnel but that seems like a lot of work.

**Comment by NathanHolcomb** [comment]
Thanks for the feedback Mike, we have similar issues with regards to the
NOS network not being IPV6 capable at this point. We are working on getting
an  external IPV6 hotspot set up so we can at least begin testing
off-network while the development continues. I didn't even think of the
impacts of your home ISP not being IPV6 capable and given the remote work
we do that could definitely pose a problem. That will definitely be
something I need to bring up at our next internal meeting.

On Tue, Jan 24, 2023 at 7:40 AM Mike Neilson ***@***.***>
wrote:

> I wouldn't expect too many problems in OpenDCS itself. There's likely a
> few places that do an address validation check that may be IPv4 specific
> but once identified those would be fairly easy to fix. But we should start
> making sure any tests of network connectivity also work with IPv6... once
> we figure out exactly how.
>
> That said, I'd be more concerned about downstream users. For example I
> still need to request an account for my home lab, but I'm on Comcast at
> home as an ISP. I don't get an IPv6 address; so my home lab may not be able
> to connect if only IPv6 connectivity is provided.
>
> I don't think we've gotten IPv6 address at the office yet either. That
> could be true of many agencies especially small partners.
>
> So other issues, this piece of text implies I control my ISP:
>
>    - All external partners will identify systems that interface with
>    networked Federal information systems and develop plans to migrate all such
>    network interfaces to the use of IPv6.
>
> I don't. That would be the FTC/FCC, and even then barely. So my plan is
> "wait for my ISP to provide said service."
> And at the office there's already a push for IPv6... been going on for
> years.
>
> Granted I could get an IPv6 tunnel but that seems like a lot of work.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/opendcs/opendcs/discussions/168#discussioncomment-4766289>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AY5A5YLSY2PHJJEHAJUMIA3WT7ESNANCNFSM6AAAAAAUFAMR6U>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Nathan Holcomb
NOAA/NOS/COOPS
Chesapeake Instrument Lab
672 Independence Parkway
Chesapeake VA 23320

Email: ***@***.***
Phone: (757) 390-9521

**Comment by cappelaere** [comment]
Hi Nathan,
Interesting question.  As NOAA transitions DCS to the cloud, and users are connected to the cloud, I would imagine that the use-cases might change drastically.
There might not be a need to connect via a socket interface…  You would be able to get your data directly from the cloud seamlessly (and process it).
Would love to have that discussion with you at some point :).
Pat.

**Comment by MikeNeilson** [comment]
Pat, 

There *always* a socket interface, it may just not be used directly. So IPv6 only can still be a concern, even with an HTTP connection.
